### PR TITLE
Update Capped Bandwidth

### PR DESCRIPTION
**Pull request by alrevuelta**
* Update Capped Bandwidth including latest RLN v2 changes and membership limit enforced by the contract.

### CODE REVIEW

File: docs/capped-bandwidth.md [hunk 1]
```diff
@@ -2,63 +2,42 @@
...
+The main problem is that one can't just choose the bandwidth it allocates to `relay`. One could set the maximum bandwidth willing to allocate to `store` but this is not how `relay` works. The required bandwidth is not set by the node, but by the network. If a pubsub topic `a` has a traffic of 50 Mbps (which is the sum of all messages being sent multiplied by its size, times the D_out degree), then if a node wants to stay up to date in that topic, and relay traffic in it, then it will require 50 Mbps. There is no thing such as "partially contributing" to the topic (with eg 25Mbps) because then you will be losing messages, becoming an unreliable peer. The network sets the pace.
```
Comment (fryorcraken) [inline docs/capped-bandwidth.md:13]:
```suggestion
The main problem is that one can't just choose the bandwidth it allocates to `relay`. One could set the maximum bandwidth willing to allocate to `store` but this is not how `relay` works. The required bandwidth is not set by the node, but by the network. If a pubsub topic `a` has a traffic of 50 Mbps (which is the sum of all messages being sent multiplied by its size, times the D_out degree), then if a node wants to stay up to date in that topic, and relay traffic in it, then it will require 50 Mbps. There is no thing such as "partially contributing" to the topic (with eg 25Mbps) because then you will be losing messages, becoming an unreliable peer and potentially be disconnected. The network sets the pace.
```

I think it's worth mentioning gossipsub's scoring meaning that being an unreliable peer actually has a drastic consequence beyond missing messages.

File: docs/capped-bandwidth.md [hunk 2]
```diff
@@ -2,63 +2,42 @@
...
-## TLDR
-
-- Waku should guarantee a capped bandwidth so that everyone can run a node.
-- The guarantee is a "statistical guarantee", since there is no way of enforcing a strict limit.
-- Current RLN approach is to rate limit 1 message every x seconds. A better approach would be x messages every day, which helps achieving such bandwidth limit.
-- To follow up: Variable RLN memberships. Eg. allow to chose tier 1 (100msg/day) tier 2 (200msg/day) etc.
\ No newline at end of file
+We can expect:
+* A traffic of `266 msg/second` on average (`160000/600`)
+* A traffic of `6 MBps` on average (266 * 4KB * 6), where `4KB` is the average message size and `6` is the average gossipsub D-out degree.
```
Comment (fryorcraken) [inline docs/capped-bandwidth.md:43]: Can you specify that it is per shard?

**Review by fryorcraken — APPROVED** [review]

File: docs/capped-bandwidth.md (hunk 2 above)
Comment (alrevuelta) [reply docs/capped-bandwidth.md]: fixed: af2101e657cfc6a336f992d6d0f38c873966a042

File: docs/capped-bandwidth.md [hunk 3]
```diff
@@ -2,63 +2,42 @@
...
-## TLDR
-
-- Waku should guarantee a capped bandwidth so that everyone can run a node.
-- The guarantee is a "statistical guarantee", since there is no way of enforcing a strict limit.
-- Current RLN approach is to rate limit 1 message every x seconds. A better approach would be x messages every day, which helps achieving such bandwidth limit.
-- To follow up: Variable RLN memberships. Eg. allow to chose tier 1 (100msg/day) tier 2 (200msg/day) etc.
\ No newline at end of file
+We can expect the following message rate and bandwidth per shard:
+* A traffic of `266 msg/second` on average (`160000/600`)
+* A traffic of `6 MBps` on average (266 * 4KB * 6), where `4KB` is the average message size and `6` is the average gossipsub D-out degree.
```
Comment (fryorcraken) [inline docs/capped-bandwidth.md:43]: I think my [previous comment](https://github.com/waku-org/research/pull/107#discussion_r1783728797) was confusing, or I am confused.

Isn't 160,000 messages per epoch for the whole network @s-tikhomirov ?
Hence, `(160 000 / 600) / 8 = 33.3 msg/second` per shard?

Comment (s-tikhomirov) [reply docs/capped-bandwidth.md:43]: I considered shards out of scope for the spec, so yes, these limits apply for the whole network.

Comment (alrevuelta) [reply docs/capped-bandwidth.md]: my bad, fixed: a76b9ef63b4c3117c828c5e3fef79d05853817e7

File: docs/capped-bandwidth.md [hunk 4]
```diff
@@ -2,63 +2,46 @@
...
+The main problem is that one can't just choose the bandwidth it allocates to `relay`. One could set the maximum bandwidth willing to allocate to `store` but this is not how `relay` works. The required bandwidth is not set by the node, but by the network. If a pubsub topic `a` has a traffic of 50 Mbps (which is the sum of all messages being sent multiplied by its size, times the D_out degree), then if a node wants to stay up to date in that topic, and relay traffic in it, then it will require 50 Mbps. There is no thing such as "partially contributing" to the topic (with eg 25Mbps) because then you will be losing messages, becoming an unreliable peer and potentially be disconnected. The network sets the pace.
```
Comment (s-tikhomirov) [inline docs/capped-bandwidth.md:13]: Can a node operator choose a lower out-degree than the recommended `D_out`? Would it be penalized in that case?

Comment (alrevuelta) [reply docs/capped-bandwidth.md:13]: Sure, its configurable. You can do that and you will lower your bandwidth consumption but your node will be less resilient.

I can't think of any penalization (from others). But its not in your best interest.

An indirect penalization could be in the gossipsub score. If you have less D, then you may not perform as good as other peers with more. For example, you will have less chances to be the first peer to deliver a message to others.

**Review by s-tikhomirov — APPROVED** [review]
